module/mailnews/newsread.c

```diff
@@ -373,11 +377,13 @@ INT_PTR CALLBACK ReadNewsDialogProc(HWND hDlg, UINT message, WPARAM wParam, LPAR
 	 /* If user posts article, rescan so that it will show up */
 	 if (UserPostArticle(hDlg, info->newsgroup, info->group_name_rsc, NULL))
 	    RequestArticles(info->newsgroup);
+   ResetListSort(hList);
```

Comment:
The formatting of these switch statements is driving me batty, but I didn't want to muck with it much. If desired, I can format them later in this PR or in a separate PR for readability.

Reply:
a separate PR would be good